packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -37,20 +50,104 @@ const COPY_ICON_CLASS = 'elyra-copy-icon';
 const INSERT_ICON_CLASS = 'elyra-add-icon';
 
 /**
- * CodeSnippetTable props.
+ * CodeSnippetDisplay props.
  */
-interface ICodeSnippetProps {
+interface ICodeSnippetDisplayProps {
   codeSnippets: ICodeSnippet[];
+  getCurrentWidget: () => Widget;
 }
 
 /**
  * A React Component for code-snippets display list.
  */
-class CodeSnippetTable extends React.Component<ICodeSnippetProps> {
+class CodeSnippetDisplay extends React.Component<ICodeSnippetDisplayProps> {
   // TODO: Use code mirror to display code
-  // TODO: implement copy to clipboard command
-  // TODO: implement insert code to file editor command (first check for code language matches file editor kernel language)
 
+  // Handle code snippet insert into an editor
+  private insertCodeSnippet(snippet: ICodeSnippet): void {
+    const widget: Widget = this.props.getCurrentWidget();
+    const snippetStr: string = snippet.code.join('\n');
+
+    if (
+      widget instanceof DocumentWidget &&
+      (widget as DocumentWidget).content instanceof FileEditor
+    ) {
+      const documentWidget = widget as DocumentWidget;
+      const fileEditor = (documentWidget.content as FileEditor).editor;
+
+      if (PathExt.extname(widget.context.path) === '.md') {
```

Comment:
while [JuyterLab current only recognizes the `.md` extension](https://github.com/jupyterlab/jupyterlab/issues/8387) for markdown. there are [other extensions commonly used for markdown](https://superuser.com/a/285878) should we perhaps check for some of those extensions as well? not suggestion we try to check for all of them but perhaps at a minimum the same set of [extensions that GitHub recognizes](https://github.com/github/markup/blob/b865add2e053f8cea3d7f4d9dcba001bdfd78994/lib/github/markups.rb#L1). just a thought